tools/tools.go

```diff
@@ -35,21 +36,30 @@ type Tools struct {
 	LastRefresh time.Time
 	Logger      func(msg string)
 	installed   map[string]string
+	mutex       sync.RWMutex
```

Comment:
I used [RWMutex](https://golang.org/pkg/sync/#RWMutex) to handle the read/write access to the map.
I think this was a better approach than using the cleaner sync [Map](https://golang.org/pkg/sync/#Map) because it does not require changes in the code logic (using functions to interact with the new Map type)